src/content/1.7/development/components/grid/tutorials/work-with-bulk-actions.md

```diff
@@ -1,6 +1,142 @@
+
+final class CustomerGridDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    // ...
+
+    protected function getBulkActions()
+    {
+        return (new BulkActionCollection())
+            ->add(
+                (new SubmitBulkAction('enable_selection'))
+                ->setName($this->trans('Enable selection', [], 'Admin.Actions'))
+                ->setOptions([
+                    'submit_route' => 'admin_customers_enable_bulk',
+                ])
+            )
+            ->add(
+                (new SubmitBulkAction('disable_selection'))
+                ->setName($this->trans('Disable selection', [], 'Admin.Actions'))
+                ->setOptions([
+                    'submit_route' => 'admin_customers_disable_bulk',
+                ])
+            )
+            ->add((new DeleteCustomersBulkAction('delete_selection'))
+                ->setName($this->trans('Delete selected', [], 'Admin.Actions'))
+                ->setOptions([
+                    'customers_bulk_delete_route' => 'admin_customers_delete_bulk',
+                ])
+            )
+        ;
+    }
+}
+```
+
+As you see, Grid stores Bulk actions in `BulkActionCollection` object. This mean that every Bulk action needed for Grid must be added to `BulkActionCollection` using `add()` method.
+
+## How to add Bulk actions from module?
+
+You can add additional Bulk actions to any Grid using hook. You have to follow these steps:
+
+1. Register hook `action{gridId}GridDefinitionModifier` where `{gridId}` is Grid id (e.g. `customer` for Customers grid, this means that hook name would be `actionCustomerGridDefinitionModifier`).
+2. Implement hook method in your module:
+
+```php
+<?php
+// modules/mymodule/mymodule.php
+
+class MyModule extends Module
+{
+    // ...
+
+    /**
+     * Use hook to add Bulk action for subscribing multiple customers to newsletter
+     */
+    public function hookActionCustomerGridDefinitionModifier(array $params)
+    {
+        // $params['definition'] is instance of \PrestaShop\PrestaShop\Core\Grid\Definition\GridDefinition
+        $params['definition']->getBulkActions()->add(
+            ->add(
+                (new SubmitBulkAction('disable_selection'))
+                    ->setName('Subscribe newsletter')
+                    ->setOptions([
+                        // in most cases submit action should be implemented by module
+                        'submit_route' => 'admin_my_module_customers_bulk_subscribe_newsletter',
+                    ])
+                ) 
+            )
+        ;
+    }
+}
+```
+
+Now you should be able to see new Bulk action available in Customers grid!
+
+## How to create custom Bulk action?
+
+PrestaShop already comes with a bunch of Bulk actions that are available for use in your Grids or for extending PrestaShop grids. However, in some use cases you may find that exisitng Bulk actions does not fit your needs. Luckly, there's a solution, you can create custom Bulk actions.
```

Comment:
```suggestion
PrestaShop already comes with a bunch of Bulk actions that are available for use in your Grids or for extending PrestaShop grids. However, in some use cases you may find that existing Bulk actions do not fit your needs. Luckily, there is a solution, you can create custom Bulk actions.
```
Please avoid "let's" or "there's" 😉I think this might not be suitable for a doc which aims to be accurate and exhaustive rather than friendly

Not sure about this though @LouiseBonnard wdyt ?

Reply:
Absolutely, please avoid abbreviated forms when writing an official text. ;-)

Reply:
Ok, but I can find these words in other documentations, like Symfony.